### ISSUE TITLE
clusterctl delete fails if provider has cluster-wide resources which require conversion

### ISSUE THREAD

**Issue by sbueringer** [kubernetes-sigs/cluster-api#5417]
**What steps did you take and what happened:**
[A clear and concise description on how to REPRODUCE the bug.]

`clusterctl delete` should work even if there are cluster-wide resources which require conversion.

This can be reproduced by doing roughly the following:
* clusterctl-v0.4.4 init --infrastructure aws
* Deploy an AWS cluster which uses one of the cluster-wide resources, e.g. `AWSClusterControllerIdentity`
* clusterctl-v1.0.0 upgrade apply --contract v1beta1
* clusterctl-v1.0.0 delete --all
```bash
Deleting Provider="infrastructure-aws" Version="v1.0.0" Namespace="capa-system"
Deleting Provider="bootstrap-kubeadm" Version="v1.0.0" Namespace="capi-kubeadm-bootstrap-system"
Error: failed to list objects for the "infrastructure.cluster.x-k8s.io/v1beta1, Kind=AWSClusterControllerIdentity" GroupVersionKind: conversion webhook for infrastructure.cluster.x-k8s.io/v1alpha4, Kind=AWSClusterControllerIdentity failed: Post "https://capa-webhook-service.capa-system.svc:443/convert?timeout=30s": service "capa-webhook-service" not found
```

I reproduced it in another way to speed it up a bit: 
```
$ clusterctl-v1.0.0 init --infrastructure aws

# Edit CRD to change storage version to v1alpha4

# Deploy
# apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
# kind: AWSClusterControllerIdentity
# metadata:
#   labels:
#     clusterctl.cluster.x-k8s.io/move-hierarchy: ""
#   name: default
# spec:
#   allowedNamespaces:
#     list: null
#     selector: {}

# Change storageVersion to v1beta1 

$ clusterctl-v1.0.0 delete --all
Deleting Provider="infrastructure-aws" Version="v1.0.0" Namespace="capa-system"
Deleting Provider="bootstrap-kubeadm" Version="v1.0.0" Namespace="capi-kubeadm-bootstrap-system"
Error: failed to list objects for the "infrastructure.cluster.x-k8s.io/v1beta1, Kind=AWSClusterControllerIdentity" GroupVersionKind: conversion webhook for infrastructure.cluster.x-k8s.io/v1alpha4, Kind=AWSClusterControllerIdentity failed: Post "https://capa-webhook-service.capa-system.svc:443/convert?timeout=30s": service "capa-webhook-service" not found
```

**What did you expect to happen:**
Should have just worked

**Anything else you would like to add:**

The problem is the following:
* clusterctl delete first deletes the CAPA provider
* clusterctl now tries to delete the bootstrap kubeadm provider, during that
  * `(p *providerComponents) Delete` calls `(k *proxy) ListResources`
  * `(k *proxy) ListResources`:
    * lists all resources matching the `clusterctl.cluster.x-k8s.io` and `cluster.x-k8s.io/provider=bootstrap-kubeadm` labels in the provider namespace (which succeeds)
    * lists all cluster-wide resources matching the same labels. This fails as apparently the conversion is triggered on the server-side before the labels are evaluated

Notes:
* Not sure how to resolve this as we don't know which CRDs belong to which provider (they are not labeled)
* Depending on the solution we should probably enhance our documentation with a note that after `clusterctl delete` the list calls might not work anymore (i.e. if they require conversion)

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

**Comment by fabriziopandini** [comment]
Another options that we can explore is to add a predicate to discovery.FilteredBy, but this requires digging a little bit into how this implemented. However, If this could avoid us to access CRD definitions one by one to check metadata, probably it is worth the effort.

**Comment by sbueringer** [comment]
I don't think we can solve it in `discovery.FilteredBy`. I don't think we have the data in the resourceList to decide which CRD belongs to which provider:
![image](https://user-images.githubusercontent.com/4662360/136824097-9dd2a4f0-07d3-4bd3-afd0-0e4c4b3ff4e8.png)

I think we probably have to use the `cluster.x-k8s.io/provider: infrastructure-aws` label on the CRDs.
